src/Datasource/RepositoryInterface.php

```diff
@@ -17,6 +17,8 @@
 /**
  * Describes the methods that any class representing a data storage should
  * comply with.
+ *
+ * @method int deleteEach($conditions, $options = []);
```

Comment:
that interface is locked. If I add a new method it breaks BC. I'm just adding a type hint for IDEs here. The best I could do for now. I would love fix this in Cake 4. We need better interface coupling for tables.